build_tools/cmake/iree_cc_binary.cmake

```diff
@@ -62,28 +62,34 @@ function(iree_cc_binary)
   # Prefix the library with the package name, so we get: iree_package_name
   iree_package_name(_PACKAGE_NAME)
   iree_package_ns(_PACKAGE_NS)
-  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+  set(_NAME_PARTS "")
+  list(APPEND _NAME_PARTS "${_PACKAGE_NAME}")
+  list(APPEND _NAME_PARTS "${_RULE_NAME}")
+  list(JOIN _NAME_PARTS "_" _NAME)
```

Comment:
This is to allow the package name to be empty? I'm surprised it works for that case and doesn't give you a leading underscore. Have you tested?

Reply:
Yep! It works as expected, but I'm sorry. Appending empty string to _an empty CMake list_ is a no-op. If the list already has elements then you can append empty strings...

https://stackoverflow.com/questions/54569185/how-to-add-empty-element-to-cmake-list

I figured joining a list of parts was somewhat intuitive, but could also go for an explicit "if x is empty, set to y, otherwise set to x_y".

Reply:
![1200px-Cmake svg](https://user-images.githubusercontent.com/5732088/169423404-5d3dad13-2776-4c0b-8977-30fc768a0443.png)

Reply:
Okay, that CMake code was trying too hard to be clever. I've switched to an explicit if/else.